lib/transport/transport-socket.h

```diff
@@ -34,6 +36,12 @@ struct _LogTransportSocket
   gint address_family;
   gint proto;
   void (*parse_cmsg)(LogTransportSocket *self, struct cmsghdr *cmsg, LogTransportAuxData *aux);
+  /* Having this in the LogTransport could be a huge advantage, and
+   * could make the proxied state handling and implementation simpler and cleaner
+   * Consider to add it to LogTransport even though that is a storage overhead, and
+   * far from optimal, as LogTransportSocketProxy actually is needed only for `LogTransportSocket`s
+   */
```

Comment:
I want to emphasize that I do see your point, however just as you said it is not optimal. 

I think that because of how special the proxied case is it should be it's own separate island (as much as possible) and should not change other (more conventional) interfaces / classes.